packages/iov-bns/src/util.ts

```diff
@@ -116,7 +116,15 @@ export function isConfirmedWithSwapClaimOrTimeoutTransaction(
   return isSwapClaimTransaction(unsigned) || isSwapTimeoutTransaction(unsigned);
 }
 
+function bnsFromOrToTag(addr: Address): string {
+  const id = Uint8Array.from([...Encoding.toAscii("wllt:"), ...decodeBnsAddress(addr).data]);
+  const key = Encoding.toHex(id).toUpperCase();
+  const value = "s"; // "s" for "set"
+  return `${key}='${value}'`;
+}
+
 export function buildTxQuery(query: BcpTxQuery): QueryString {
+  const addressConponents = query.address !== undefined ? [bnsFromOrToTag(query.address)] : [];
   const tagComponents = query.tags !== undefined ? query.tags.map(tag => `${tag.key}='${tag.value}'`) : [];
   // In Tendermint, hash can be lower case for search queries but must be upper case for subscribe queries
   const hashComponents = query.id !== undefined ? [`tx.hash='${query.id}'`] : [];
```

Comment:
Reading this code and remembering the bug, I am surprised that you do not auto-uppercase the tx.hash to avoid those annoying issues

Reply:
This is because the BCP representation (type `TransactionId`) of a Tendermint transaction hash is uppercase hex. non-uppercase hex is corrupted data. When there is corrupted data, we should never hide the corruption by auto-correction but instead find the root of the corruption.

We cannot allow mixed hex case in `TransactionId` as we're comparing BCP `TransactionId` by string comparison and `'aabb' != 'AABB'`.

Reply:
Okay, so this is handled above, and any errors here mean mis-use of the api (force asserting garbage as TransactionId). Fair enough

Reply:
The only thing we could do here is check the format of `TransactionId` and error out if broken. However, since `TransactionId` always comes from up, I want to avoid checking it in every single API unless necessary.